test/unit/sourceset.test.js

```diff
@@ -0,0 +1,506 @@
+/* eslint-env qunit */
+import videojs from '../../src/js/video.js';
+import document from 'global/document';
+import window from 'global/window';
+import log from '../../src/js/utils/log.js';
+import sinon from 'sinon';
+
+const wait = 100;
+let qunitFn = 'module';
+
+if (videojs.browser.IS_IE8) {
+  qunitFn = 'skip';
+}
+
+// if we cannot overwrite the src property, there is no support
+try {
+  const el = document.createElement('video');
+
+  Object.defineProperty(el, 'src', {
+    get() {},
+    set() {}
+  });
+} catch (e) {
+  qunitFn = 'skip';
+}
+
+const Html5 = videojs.getTech('Html5');
+const oldMovingMedia = Html5.prototype.movingMediaElementInDOM;
+
+const validateSource = function(assert, player, sources, checkMediaElSource = true) {
+  const tech = player.tech_;
+  const mediaEl = tech.el();
+
+  // assert.deepEqual(player.currentSource(), sources[0], 'currentSource is correct');
```

Comment:
Once we actually change the source cache based on a sourceset we will do this.

Reply:
Is this change made in https://github.com/videojs/video.js/pull/4717? Can we remove it from here then?

Reply:
yeah I think we can remove it from here